tests/integration/test_metric_collection.py

```diff
@@ -0,0 +1,101 @@
+        response = None
+        try: 
+            response = requests.get(url)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            print(f"An error occurred while making the request: {e}")
+
+        if response is not None and response.status_code == 200:
+            processed_successfully = True
+            try:
+                print("Successfully downloaded!")
+                lines = response.content.splitlines()
+                metrics = [json.loads(line) for line in lines]
+                merged_json = merge_jsons(metrics)
+
+                actual_json = json.dumps(merged_json, sort_keys=True, indent=2)
+                expected_json = json.dumps(expected_json_raw, sort_keys=True, indent=2)
+
+                write_actual = os.getenv("WRITE_ACTUAL_JSON", "False")
+                if write_actual == "True":
+                    with open("actual.json", "w", newline='\n') as f:
+                        f.write(actual_json)
+
+                length_matches = False
+                if len(actual_json.splitlines()) == len(expected_json.splitlines()):
+                    print(
+                        f'LineCount of outputs matches, expected: {len(expected_json.splitlines())}, actual: {len(actual_json.splitlines())}')
+                    length_matches = True
+                else:
+                    print(
+                        f'LineCount of outputs does not match, expected: {len(expected_json.splitlines())}, actual: {len(actual_json.splitlines())}')
+
+                metric_names = get_unique_metric_names(merged_json)
+                metric_matches = False
+                if all(name in metric_names for name in expected_metric_names):
+                    print("All specific metric names are found in the response.")
+                    metric_matches = True
+                else:
+                    missing_metric_names = [name in metric_names for name in expected_metric_names]
+                    print('Some specific metric names are not found in the response')
+                    print(f'Missing metrics: {missing_metric_names}')
+            except Exception as e:
+                print('An exception occurred: {}'.format(e))
+                processed_successfully = False;
+
+
+            if processed_successfully and metric_matches and length_matches:
+                break
+            else:
+                print('Retrying...')
+                time.sleep(2)
+        else:
+            if response is not None:
+                print('Failed to download metrics. Response code:',
+                    response.status_code)
+            print('Retrying...')
+            time.sleep(2)
+
+    if time.time() - start_time >= timeout:
+        if not metric_matches and actual_json is not None:
+            print(f'Actual json:')
+            print(actual_json)
+
+        raise ValueError("Timed out waiting for specific metric names")
```

Comment:
It would be maybe good to print these "indicators: processed_successfully and metric_matches and length_matches" here to simplify investigation.

Reply:
added